### PR TITLE
Add 'IF NOT EXISTS' to prevent duplicate table exception

### PR DESCRIPTION
**Pull request by ant1j**
PR for issue #2060: Avoid exception handling for PostgresTarget.create_marker_table() with "CREATE TABLE IF NOT EXISTS"

## Description
adding Postgresq `IF NOT EXISTS` clause in order to avoid having to rely on a exception handling when the `marker_table` already exists

## Motivation and Context
Fix issue #2060 

## Have you tested this? If so, how?
Not tested directly, but the SQL query is valid for Postgresql since version 9.1

### CODE REVIEW

File: luigi/contrib/postgres.py [hunk 1]
```diff
@@ -207,24 +207,21 @@ def create_marker_table(self):
...
         else:
-            sql = """ CREATE TABLE {marker_table} (
+            sql = """ CREATE TABLE IF NOT EXISTS {marker_table} (
                       update_id TEXT PRIMARY KEY,
                       target_table TEXT,
                       inserted TIMESTAMP);
                   """.format(marker_table=self.marker_table)
         try:
             cursor.execute(sql)
         except psycopg2.ProgrammingError as e:
```
Comment (dlstadther) [inline luigi/contrib/postgres.py]: Flake8 doesn't like the assignment, but non-use of `e` here.

I think we can just remove the try/except here for `cursor.execute(sql)`.

**Comment by ant1j** [comment]
I have updated the code; it seems that the PR is automatically updated and tests are run also, which means I have nothing more to do... is that right?

**Comment by Tarrasch** [comment]
Can you find somebody to review this code? Somebody who knows about the technologies (postgres?) this PR touches upon.

**Comment by ant1j** [comment]
Isn't the `luigi/test/postgres_with_server_test.py` file supposed to do so?

I am more fluent in (postgre)sql than Python, so I know the query is working. I crafted quickly 2 sqlfiddle here and they do not return any error:
- http://sqlfiddle.com/#!15/d1587 : `CREATE TABLE IF NOT EXISTS` with values inserted => OK
- http://sqlfiddle.com/#!15/b33421 : (`CREATE TABLE IF NOT EXISTS` twice and then values inserted => OK

I don't know if it is enough as a code review; maybe some people using Redshift could also confirm that, as Redshift as the same feature (see http://docs.aws.amazon.com/fr_fr/redshift/latest/dg/r_CREATE_TABLE_NEW.html)

**Comment by Tarrasch** [comment]
I think it helps to have real humans look at code. Also you'll learn more if you get feedback from peers. :)

If however noone volunteers for a few days or a week. I'm fine with just merging this.

**Comment by smehan** [comment]
I'll have a look at it today.

-s.

**Comment by dlstadther** [comment]
If no one else reviews, I'll merge this soon. Thanks

**Pull request merged**